### ISSUE TITLE
Firefox freezes for multiple seconds while addon is installed

### ISSUE THREAD

**Issue by syssgx** [mozilla/r2d2b2g#83]
OK, you might want to file a new bugzilla issue on this:
when downloading r2d2b2g and clicking the install button, browsers freezes for like 10 seconds. Same things happens if you remove it through the addons tab (after closing it).

My guess is that the installation system is not designed to handle such big addons, so we might want to have a deeper look at it.

**Comment by mykmelez** [comment]
Yes, good point. In fact sometimes it hangs so long that the slow-running script dialog appears! I suspect that the time is being spent unpacking the addon, which we have to do in order to start B2G Desktop and give it access to its profile.

It should be possible to unpack addons asynchronously. We should file a Firefox bug on this hang.

**Comment by syssgx** [comment]
cool. :)
are you going to file the a new bug on this?

**Comment by mykmelez** [comment]
Yes, I'll file one as soon as I get a screenshot of the slow-running script dialog!

**Comment by mykmelez** [comment]
I filed this as [bug 814505](https://bugzilla.mozilla.org/show_bug.cgi?id=814505).

**Comment by mykmelez** [comment]
Note that the Simulator addon will never be small. A zipped Gaia v1-train profile with only core apps and no additional locales is currently 34MB, and there's not much we can do to make it smaller.

So even when we get the Simulator running on Firefox's XULRunner and stop shipping any SDK modules with the addon, such that the Simulator and addon parts of the package can be measured in the 10s to 100s of KB, the addon will still be at least 34MB.

Which means that we really do need Firefox to unpack addons asynchronously or do the unpacking ourselves after installation (adding complexity to the implementation and taking a hit on the user experience, since a core benefit of the experience today is that there is only a single installation step).

**Comment by jryans** [comment]
@mykmelez: Is this a duplicate of issue #28?

**Comment by mykmelez** [comment]
> @mykmelez: Is this a duplicate of issue #28?

It's actually a different issue, as this one is about addon installation, while that one is about the addon load time during Firefox startup.
